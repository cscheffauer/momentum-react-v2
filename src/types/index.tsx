import type { Button, Dialog } from '@momentum-design/components/dist/react';

export type MdcButtonProps = React.ComponentProps<typeof Button>;
export type MdcDialogProps = React.ComponentProps<typeof Dialog>;
