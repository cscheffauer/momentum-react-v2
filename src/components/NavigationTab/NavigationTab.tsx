import React, { FC } from 'react';
import classnames from 'classnames';
import FocusRing from '../FocusRing';

import Icon from '../Icon';
import Badge from '../Badge';
import Text from '../Text';
import ButtonSimple from '../ButtonSimple';

import { DEFAULTS, STYLE } from './NavigationTab.constants';
import { Props } from './NavigationTab.types';
import './NavigationTab.style.scss';

const NavigationTab: FC<Props> = (props: Props) => {
  const {
    icon,
    label,
    count = DEFAULTS.COUNT,
    className,
    id,
    size,
    style,
    active,
    ...otherProps
  } = props;

  const iconComponent = icon ? (
    <Icon className={STYLE.icon} name={icon} scale={24} weight={'filled'} />
  ) : null;

  const labelComponent =
    size == 200 && label ? (
      <Text className={STYLE.label} type="subheader-secondary" tagName="h3">
        {label}
      </Text>
    ) : null;

  const badgeComponent =
    count > 0 ? (
      <Badge
        className={STYLE.count}
        type="counter"
        counter={count}
        maxCounter={99}
        overlay={size !== 200}
      />
    ) : null;

  return (
    <FocusRing>
      <ButtonSimple
        className={classnames(STYLE.wrapper, className)}
        data-active={active || DEFAULTS.ACTIVE}
        data-size={size || DEFAULTS.SIZE}
        id={id}
        style={style}
        {...otherProps}
      >
        {iconComponent}
        {labelComponent}
        {badgeComponent}
      </ButtonSimple>
    </FocusRing>
  );
};

export default NavigationTab;
