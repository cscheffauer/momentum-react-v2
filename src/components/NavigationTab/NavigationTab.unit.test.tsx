import React from 'react';
import { mount } from 'enzyme';

import NavigationTab, { NAVIGATION_TAB_CONSTANTS as CONSTANTS } from './';
import { STYLE, SIZES } from './NavigationTab.constants';
import { NavTabSize } from './NavigationTab.types';
import { mountAndWait } from '../../../test/utils';

describe('<NavigationTab />', () => {
  describe('snapshot', () => {
    it('should match snapshot', async () => {
      expect.assertions(1);

      const container = await mountAndWait(<NavigationTab />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = await mountAndWait(<NavigationTab className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountAndWait(<NavigationTab id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = await mountAndWait(<NavigationTab style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size', async () => {
      expect.assertions(1);

      const sizes = Object.values(SIZES).map((size, index) => {
        return <NavigationTab key={index} size={size as NavTabSize} />;
      });
      const container = await mountAndWait(<div>{sizes}</div>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with label', async () => {
      expect.assertions(1);

      const label = 'Example label';

      const container = await mountAndWait(<NavigationTab label={label} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with icon', async () => {
      expect.assertions(1);

      const icon = 'contacts';

      const container = await mountAndWait(<NavigationTab icon={icon} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with count', async () => {
      expect.assertions(1);

      const count = 100;

      const container = await mountAndWait(<NavigationTab count={count} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with active', async () => {
      expect.assertions(1);

      const active = true;

      const container = await mountAndWait(<NavigationTab active={active} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size, label, icon', async () => {
      expect.assertions(1);

      const size = 48;
      const label = 'Contacts';
      const icon = 'contacts';

      const container = await mountAndWait(<NavigationTab size={size} label={label} icon={icon} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size, label, icon, count, active', async () => {
      expect.assertions(1);

      const size = 48;
      const label = 'Contacts';
      const icon = 'contacts';
      const count = 0;
      const active = false;

      const container = await mountAndWait(
        <NavigationTab size={size} label={label} icon={icon} count={count} active={active} />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      expect.assertions(1);

      const wrapper = await mountAndWait(<NavigationTab />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.classList.contains(STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const wrapper = await mountAndWait(<NavigationTab className={className} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const wrapper = mount(<NavigationTab id={id} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const wrapper = await mountAndWait(<NavigationTab style={style} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided size when size is provided', async () => {
      expect.assertions(1);

      const size = CONSTANTS.DEFAULTS.SIZE;

      const wrapper = await mountAndWait(<NavigationTab size={size} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('data-size')).toBe(`${size}`);
    });

    it('should have provided label when label is provided and size is 200 (size 48 has no label)', async () => {
      expect.assertions(1);

      const label = 'Messaging';

      const wrapper = await mountAndWait(<NavigationTab size={200} label={label} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      const target = element.getElementsByClassName(STYLE.label)[0];

      expect(target.innerHTML).toBe(label);
    });

    it('should have provided active when active is provided', async () => {
      expect.assertions(1);

      const active = CONSTANTS.DEFAULTS.ACTIVE;

      const wrapper = await mountAndWait(<NavigationTab active={active} />);
      const element = wrapper.find(NavigationTab).getDOMNode();

      expect(element.getAttribute('data-active')).toBe(`${active}`);
    });
  });

  describe('actions', () => {
    it('should handle mouse press events', async () => {
      expect.assertions(1);

      const mockCallback = jest.fn();

      const wrapper = await mountAndWait(<NavigationTab onPress={mockCallback} />);
      const component = wrapper.find(NavigationTab);

      component.props().onPress({
        type: 'press',
        pointerType: 'mouse',
        shiftKey: false,
        ctrlKey: false,
        metaKey: false,
        target: component.getDOMNode(),
        altKey: false,
      });

      expect(mockCallback).toBeCalledTimes(1);
    });
  });
});
