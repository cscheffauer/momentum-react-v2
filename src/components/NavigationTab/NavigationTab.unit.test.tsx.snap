// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationTab /> snapshot should match snapshot 1`] = `
<NavigationTab>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with active 1`] = `
<NavigationTab
  active={true}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={true}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={true}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with className 1`] = `
<NavigationTab
  className="example-class"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper example-class"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper example-class md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with count 1`] = `
<NavigationTab
  count={100}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Badge
                className="md-navigation-tab-count"
                counter={100}
                maxCounter={99}
                overlay={true}
                type="counter"
              >
                <Badge
                  className="md-navigation-tab-count"
                  counter={100}
                  maxCounter={99}
                  overlay={true}
                  type="counter"
                >
                  <mdc-badge
                    class="md-navigation-tab-count"
                    suppressHydrationWarning={true}
                  />
                </Badge>
              </Badge>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with icon 1`] = `
<NavigationTab
  icon="contacts"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Mrv2Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                weight="filled"
              >
                <Icon
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                  data-autoscale={false}
                  data-scale={24}
                  data-test="contacts"
                  name="contacts-filled"
                  style={Object {}}
                >
                  <mdc-icon
                    class="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    style={Object {}}
                    suppressHydrationWarning={true}
                  />
                </Icon>
              </Mrv2Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with id 1`] = `
<NavigationTab
  id="example-id"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        id="example-id"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              id="example-id"
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with label 1`] = `
<NavigationTab
  label="Example label"
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size 1`] = `
<div>
  <NavigationTab
    key="0"
    size={40}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={40}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={40}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
  <NavigationTab
    key="1"
    size={48}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={48}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={48}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
  <NavigationTab
    key="2"
    size={200}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <ButtonSimple
          className="md-navigation-tab-wrapper"
          data-active={false}
          data-size={200}
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <FocusRing>
            <FocusRing
              focusClass="md-focus-ring-wrapper children"
              focusRingClass="md-focus-ring-wrapper children"
            >
              <button
                className="md-navigation-tab-wrapper md-button-simple-wrapper"
                data-active={false}
                data-size={200}
                onBlur={[Function]}
                onClick={[Function]}
                onDragStart={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                type="button"
              />
            </FocusRing>
          </FocusRing>
        </ButtonSimple>
      </FocusRing>
    </FocusRing>
  </NavigationTab>
</div>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size, label, icon 1`] = `
<NavigationTab
  icon="contacts"
  label="Contacts"
  size={48}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Mrv2Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                weight="filled"
              >
                <Icon
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                  data-autoscale={false}
                  data-scale={24}
                  data-test="contacts"
                  name="contacts-filled"
                  style={Object {}}
                >
                  <mdc-icon
                    class="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    style={Object {}}
                    suppressHydrationWarning={true}
                  />
                </Icon>
              </Mrv2Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with size, label, icon, count, active 1`] = `
<NavigationTab
  active={false}
  count={0}
  icon="contacts"
  label="Contacts"
  size={48}
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Mrv2Icon
                className="md-navigation-tab-icon"
                name="contacts"
                scale={24}
                weight="filled"
              >
                <Icon
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                  data-autoscale={false}
                  data-scale={24}
                  data-test="contacts"
                  name="contacts-filled"
                  style={Object {}}
                >
                  <mdc-icon
                    class="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink md-navigation-tab-icon"
                    data-autoscale={false}
                    data-scale={24}
                    data-test="contacts"
                    style={Object {}}
                    suppressHydrationWarning={true}
                  />
                </Icon>
              </Mrv2Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;

exports[`<NavigationTab /> snapshot should match snapshot with style 1`] = `
<NavigationTab
  style={
    Object {
      "color": "pink",
    }
  }
>
  <FocusRing>
    <FocusRing
      focusClass="md-focus-ring-wrapper children"
      focusRingClass="md-focus-ring-wrapper children"
    >
      <ButtonSimple
        className="md-navigation-tab-wrapper"
        data-active={false}
        data-size={48}
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "color": "pink",
          }
        }
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              className="md-navigation-tab-wrapper md-button-simple-wrapper"
              data-active={false}
              data-size={48}
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "color": "pink",
                }
              }
              type="button"
            />
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </FocusRing>
  </FocusRing>
</NavigationTab>
`;
